apps/user/services/authentication.go

```diff
@@ -50,6 +52,57 @@ func (service AuthenticationService) Login(c *gin.Context) map[string]string {
 	}
 }
 
+// BasicLogin ...
+func (service AuthenticationService) BasicLogin(c *gin.Context) (
+	userResponse *userSerializers.UserAuthSerializer,
+	status int,
+	err error) {
+
+	var userData userSerializers.UserLogin
+	err = c.BindJSON(&userData)
+	if err != nil {
+		return nil, http.StatusInternalServerError, err
+	}
+
+	gormDB := service.DB
+	userResponse = new(userSerializers.UserAuthSerializer)
+
+	err = gormDB.Model(&userModels.User{}).
+		Where("email = ?", userData.Email).
+		Scan(&userResponse).Error
+	if err != nil {
+		if err == gorm.ErrRecordNotFound {
+			return nil, http.StatusNotFound, errors.New("username or password is incorrect")
+		}
+		return nil, http.StatusInternalServerError, err
+	}
+	userResponse.Password, err = DecryptPassword(userResponse.Password)
+	if err != nil {
+		return nil, http.StatusInternalServerError, err
+	}
+	if userData.Password != userResponse.Password {
+		return nil, http.StatusNotFound, errors.New("username or password is incorrect")
+	}
+
+	session := sessions.Default(c)
```

Comment:
We are doing exactly the same thing in OAuth, can't we just reuse it?

Reply:
done.